### PR TITLE
[SPARK-16266][SQL][STREAING] Moved DataStreamReader/Writer from pyspark.sql to pyspark.sql.streaming 

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?
- Moved DataStreamReader/Writer from pyspark.sql to pyspark.sql.streaming to make them consistent with scala packaging
- Exposed the necessary classes in sql.streaming package so that they appear in the docs
- Added pyspark.sql.streaming module to the docs
## How was this patch tested?
- updated unit tests.
- generated docs for testing visibility of pyspark.sql.streaming classes.

### CODE REVIEW

**Comment by tdas** [comment]
@zsxwing Can you take a look?

File: python/pyspark/sql/streaming.py [hunk 1]
```diff
@@ -224,6 +226,494 @@ def _to_java_trigger(self, sqlContext):
             self.interval)
 
 
+class DataStreamReader(OptionUtils):
```
Comment (rxin) [inline python/pyspark/sql/streaming.py]: can we create a separate file? this file is already very long.

Comment (tdas) [inline python/pyspark/sql/streaming.py]: then probably we have to create directory streaming and move all the code in streaming.py into separate files. I wanted to minimize the changes. we can do the bookkeeping later. what do you think?

**Comment by zsxwing** [comment]
LGTM. Merging to master and 2.0. Thanks!

**Pull request closed without merging**